### ISSUE TITLE
Better integration with system

### ISSUE THREAD

**Issue by LukeMech** [Vanilla-OS/apx#86]
Is there any way of getting `apx` more integrated with system? Containerized `spotify-edge` from `aur` can't open login buttons using any browser from any other container or flatpak, only using installed from `aur` container. And browser installed from any container can't be chosen as default in gnome settings app.

**Comment by kbdharun** [comment]
> Is there any way of getting apx more integrated with system? Containerized spotify-edge from aur can't open login buttons using any browser from any other container or flatpak, only using installed from aur container

Yeah, this is a limitation, as applications installed inside the container (as far as I remember) can't access the browser outside it/ installed inside another container.

> And browser installed from any container can't be chosen as default in gnome settings app.

Regarding this it isn't true, browsers and applications installed inside containers can indeed be set as defaults and can be accessed by applications installed in the host/ within the same container. I have attached a demo video demonstrating this with the Vanilla control center in the host.

[apx_browser_default_demo.webm](https://user-images.githubusercontent.com/26346867/214338183-d3a938da-3fb8-4445-9374-8b14de96925e.webm)

**Comment by LukeMech** [comment]
> Regarding this it isn't true, browsers and applications installed inside containers can indeed be set as defaults and can be accessed by applications installed in the host/ within the same container. I have attached a demo video demonstrating this with the Vanilla control center in the host.

Hmm, for me firefoxes installed from fedora and arch containers aren't displayed in default apps page, kinda strange

**Comment by kbdharun** [comment]
> > Regarding this it isn't true, browsers and applications installed inside containers can indeed be set as defaults and can be accessed by applications installed in the host/ within the same container. I have attached a demo video demonstrating this with the Vanilla control center in the host.
> 
> Hmm, for me firefoxes installed from fedora and arch containers aren't displayed in default apps page, kinda strange

For Arch, I too haven't seen it in Default settings (Going to try again now) maybe due to the binary (bin) versions of the package I think. But for Fedora, it should work. Try `apx install --dnf firefox` and maybe logout and login back for it to show.

**Comment by LukeMech** [comment]
> But for Fedora, it should work. Try `apx install --dnf firefox` and maybe log out and log in back for it to show.

I tried using Fedora container (even after fresh Vanilla reinstall) and sadly, it isn't working. Restarting laptop also doesn't change anything.

**Comment by Thatoo** [comment]
Is there any plan to be able to install apps thanks to `apx install` with a GUI instead of CLI?
Going into the Software app, the only source to install from is Flathub (Flatpak) for now.
I can't see the list given in Apx : ame, apk, apt, dnf, zypper...
I can't go on Software repository neither from the Software app.

**Comment by kbdharun** [comment]
> Is there any way of getting `apx` more integrated with system? Containerized `spotify-edge` from `aur` can't open login buttons using any browser from any other container or flatpak, only using installed from `aur` container. And browser installed from any container can't be chosen as default in gnome settings app.

This issues has been fixed as we have updated `apx`'s distrobox version, now it invokes `xdg-open` to open the login screen in the correct default browser (Flatpak).
